test/unit/network/libp2p/rpc.test.ts

```diff
@@ -0,0 +1,153 @@
+import {assert} from "chai";
+import BN from "bn.js";
+import promisify from "promisify-es6";
+
+import {RpcController} from "../../../../src/network/libp2p/rpcController";
+
+import {createNode} from "./util";
+import {NodejsNode} from "../../../../src/network/libp2p/nodejs";
+import {Method} from "../../../../src/network/codec";
+import {Hello} from "../../../../src/types";
+
+const multiaddr = "/ip4/127.0.0.1/tcp/0";
+
+describe("[network] rpc", () => {
+  let nodeA: NodejsNode, nodeB: NodejsNode,
+    rpcA: RpcController, rpcB: RpcController;
+  beforeEach(async () => {
+    // setup
+    nodeA = await createNode(multiaddr);
+    nodeB = await createNode(multiaddr);
+    await Promise.all([
+      promisify(nodeA.start.bind(nodeA))(),
+      promisify(nodeB.start.bind(nodeB))(),
+    ]);
+    rpcA = new RpcController(nodeA);
+    rpcB = new RpcController(nodeB);
+    await Promise.all([
+      rpcA.start(),
+      rpcB.start(),
+    ]);
+  });
+  afterEach(async () => {
+    // teardown
+    await Promise.all([
+      rpcA.stop(),
+      rpcB.stop(),
+    ]);
+    await Promise.all([
+      promisify(nodeA.stop.bind(nodeA))(),
+      promisify(nodeB.stop.bind(nodeB))(),
+    ]);
+  });
+  it("creates a peer when when new libp2p peers are added", async function () {
+    this.timeout(3000);
+    await promisify(nodeA.dial.bind(nodeA))(nodeB.peerInfo);
+    try {
+      await new Promise((resolve, reject) => {
+        const t = setTimeout(reject, 2000);
+        rpcA.once("peer:connect", () => {
+          clearTimeout(t);
+          resolve();
+        });
+      });
+    } catch (e) {
+      assert.fail("no peer connected");
+    }
+  });
+  it("can list peers", async function () {
+    this.timeout(3000);
```

Comment:
maybe move those tests to e2e. It will slow down unit tests dramatically

Reply:
The tests actually go very fast (all network tests combined finish in much less than a second)

Reply:
Why timeout than? :D

Reply:
its for the fallback case that doesn't happen in the passing test. 
A "peer:connect" event handler and a setTimeout are set at the same time, with the timeout being a failure case. Perhaps I can reduce the setTimeout to not need the timeout extension.